pkg/api/message/v1/context/context.go

```diff
@@ -61,3 +67,22 @@ func parseVersionHeaderValue(vals []string) (name string, version string, full s
 	}
 	return
 }
+
+func (ri *requesterInfo) isSupportedClient() bool {
+	// Only version-gate on production
+	if os.Getenv("ENV") != "production" {
+		return true
+	}
+	// Err on the side of caution for unknown versions
+	if ri.LibxmtpVersion == "" || !semver.IsValid(ri.LibxmtpVersion) {
+		return true
+	}
+	if ri.LibxmtpVersion == "1.2.0-dev" {
```

Comment:
Are this and line 83 not equivalent with the semver library we are using?

Reply:
Ah, someone had mentioned earlier that prerelease versions came out smaller. But that's not what the spec says now that I checked. Makes sense to remove this and confirm that it's not needed via telemetry.